packages/dds/sequence/docs/intervalCollectionQuerying.md

```diff
@@ -0,0 +1,240 @@
+If we want to do this while still providing a typesafe API for the application author, we'd need to make the type of "indexes" above generic.
+This gets pretty involved. Something like [this](https://www.typescriptlang.org/play?ts=4.7.4#code/JYOwLgpgTgZghgYwgAgJLmgNzgG3QEwgA8AeAFXUimxwD5kBvAWAChl3k598AKUKmgC5kFDNVwBKYZgD2wfAG5WHZFAgBbGZgh8xQkZSyTpcxawC+rVgHpryACIBXdeoCeyMK4AOEAM6tPH2QAZQgAR0cIECRDcRxkAF5kEGcAI2glFn5oeCRkAGEcYCiwRksssVyUVAAJCBwfKF8yqxZA6r1cAmIAMUQwGShXcliaeiSeBCKS4ULi8AAaZAALesbfYVq16F8JRPpRrpBCUlEBXFpW9rROvGPiACUIAHNgXzAhkdvx5CeuGRAOFcAEEoFA4MM1P9Ae4ANrvKCgZ5LQ53E59BADT5nIx0AC6lxYrCmcF8zQA8tooDg4F4vEjUd1Tqj6MB1F4cBoSs1Gfdmd9GMoOF5HKkighkG4AIqRIbAgAKqB4UmSaWgjFUEDAjigIGQAFYFMhyioRWLgBKuLw9gwTcLReLNZptMqWixyqwYI5omBgADkAgoZBKdAaXSGbcmZNpuBZjGwEtVg0dps6smmiqQ9TafSQM9eScSKEIlEYgLmGwOGptbrkhAAO7ILNh3P5yN85WZD0sBAA96a14I9xJWGwgDkWlDOaRY6WgYgcGDVJbEfOaOIeLxnGavZA72JfdKqU27ZOT0HH2Gxci0QgLMSA7el8yNjsZDgAGs-B5VgHD3BwEfIcPG8FA4FSScPBkH8UGuBhkEnbNwzzYRm2nPMC2IItwhvMs13ocwADoAlAm4zwgXsoHwHooBkdRzyfbFUSWURyIvIZkGISBjh5U9HheRjhhxOJaB+CsVFhABpZBQGQL9XBkGADD5BjgNJZBpK4qJ8GaFJ1HSKBkAAfmUtjBKkvFYQABi3YQQAgKk8WEJ4axAMhQK+Myhws2EAEYCS7VoSTJG413yGQcE5TE-RAL41xYplVMvTiiG4nTQtxRKBKHOLcVEwVK3YUATj8E9yMo6jaPo7LL1yuIEpUmqhh+b1CBgUAIHwbdOBAVxAqJFgvR9GLkGeLVUXCyKKN9AESCFdhWP49j3C0niMriLLluwktbxZVhaB4ebkBpdIcGEBEkQWI7iuIUrTKWwTWBVCaIqimbYuvUs71uBqvMvehxKrLUdT1VqIHa+yuvUgC+osILDz-Kbov9JIxrAF6kfengACJexcblsaWNRlokTJe0xmLCJuog-EIxCVzzQjpVlEFFVYIA) works in typescript 4.7 or later while retaining a relatively natural API, but is a bit of a mouthful.
+It's tempting to use index types, but they don't really behave in a typesafe fashion for this kind of problem.
+
+We could always opt for a less typesafe API but keep the same shape, dropping the generics and expecting consumers to cast to appropriate query-supporting interfaces.
+
+The other bit of making indexes customizable is determining where to allow application authors to inject that piece of customization.
+There are a couple natural options:
+
+1. Configuration in `SharedStringFactory` scoped to all interval collections on that shared string
+2. An extra parameter to `getIntervalCollection`
+
+The first option is probably more convenient for most applications.
+It might be suboptimal for applications with multiple types of interval collections which have different needs.
+The second option would be better suited to such consumers.
+However, it raises some implementation questions, since applications can call `getIntervalCollection` multiple times for the same collection with a different set of indexes.
+I would lean toward going with the first option, though the issues with the second are overcomeable.
+
+In the typescript sample above, index factories are shown as taking a `Client` and `IIntervalHelpers` on construction.
+We may want to tweak this slightly, but it is all that's currently required.
+
+### Supporting segoff-based queries
```

Comment:
this would be good for the operation apis, and basically all our shared string apis.